tests/Sinch.Tests/Core/HttpTests.cs

```diff
@@ -191,5 +196,99 @@ public async Task AddOwnHeaders()
 
             _httpMessageHandlerMock.VerifyNoOutstandingExpectation();
         }
+
+        [Fact]
+        public async Task UnauthorizedIfExpiredHeaderIsNotPresent()
```

Comment:
The test name needs to highlight the fact that no call to the auth server will be made to request a new token as per new logic introduced in the code above.

Reply:
I think it's better just to add token